tools/xabuild/XABuildPaths.cs

```diff
@@ -180,15 +197,19 @@ public XABuildPaths ()
 					NuGetProps   = Path.Combine (nuget, "Microsoft.NuGet.props");
 				}
 				NuGetRestoreTargets = Path.Combine (MSBuildBin, "NuGet.targets");
+				NuGetSdkResolverPath = Path.Combine (MSBuildBin, "Microsoft.Build.NuGetSdkResolver.dll");
 				if (!File.Exists (NuGetRestoreTargets) && !string.IsNullOrEmpty (DotNetSdkPath)) {
 					NuGetRestoreTargets = Path.Combine (DotNetSdkPath, "..", "NuGet.targets");
+					NuGetSdkResolverPath = Path.Combine (DotNetSdkPath, "..", "Microsoft.Build.NuGetSdkResolver.dll");
 				}
 			}
 
 			FrameworksDirectory       = Path.Combine (prefix, "xbuild-frameworks");
 			MSBuildExtensionsPath     = Path.Combine (prefix, "xbuild");
 			MonoAndroidToolsDirectory = Path.Combine (prefix, "xbuild", "Xamarin", "Android");
-			MSBuildExeTempPath        = Path.GetTempFileName ();
+			MSBuildTempPath           = Path.Combine (Path.GetTempPath (), Path.GetRandomFileName ());
+			MSBuildExeTempPath        = Path.Combine (MSBuildTempPath, Path.GetRandomFileName ());
+			SdkResolverConfigPath     = Path.Combine (MSBuildTempPath, "SdkResolvers", "Microsoft.Build.NuGetSdkResolver", "Microsoft.Build.NuGetSdkResolver.xml");
```

Comment:
I would suggest two things:

 1. If not already done -- I didn't check -- we should ensure that the files we create in `MSBuildTempPath` are deleted on process exit, if possible, and
 2. For the "multiple `xabuild.exe` invocations in parallel" case -- and for general sanity -- we probably don't want to create `%TEMP%\SdkResolvers\...`, but instead `%TEMP%\xabuild.(SomeRandomStuff)\SdkResolvers\...`.

Reply:
This PR should be doing that already, I think.

Before:
* `%TEMP%\GetRandomFileName`
* `%TEMP%\GetRandomFileName.config`

Now:
* `%TEMP%\GetRandomFileName\GetRandomFileName`
* `%TEMP%\GetRandomFileName\GetRandomFileName.config`
* `%TEMP%\GetRandomFileName\SdkResolvers\Microsoft.Build.NuGetSdkResolvers\Microsoft.Build.NuGetSdkResolvers.xml`

Now we can just delete the entire `%TEMP%\GetRandomFileName` directory when `xabuild.exe` exits.